src/opt.rs

```diff
@@ -148,19 +151,19 @@ pub enum CastCommand {
         #[structopt(name = "TEAM")]
         /// Team abbreviation
         team_abbrev: String,
-        #[structopt(name = "CHROMECAST IP")]
+        #[structopt(name = "CHROMECAST IP", parse(try_from_str))]
         /// IP of the Chromecast
-        cast_ip: String,
+        cast_ip: Ipv4Addr,
         #[structopt(long)]
         /// If live, restart the stream from the beginning and cast the entire thing
         restart: bool,
         #[structopt(long, parse(try_from_str))]
         /// Specify the feed type to cast. Will default to supplied
         /// team's applicable Home / Away feed
         feed_type: Option<FeedType>,
-        #[structopt(long)]
+        #[structopt(long, parse(try_from_str))]
         /// Proxy server address to be passed to Streamlink
-        proxy: String,
+        proxy: Option<Uri>,
```

Comment:
I guess I somehow made `--proxy` a required option. Maybe the "try" in your `try_from_str` fixes that? Idk, this structopt thing seems like it kinda just guesses what you want it to do.

Reply:
It's required if not specified as Option.  If someone passes the value on cli, that option will be `Some(value)`, otherwise it'll be `None`.  The try_from_str is a trait that is implemented for the Uri struct, that will try to convert from str to Uri, but if it fails, will return an error.  Structopt will take those failures and exit the program, asking the user to input a correct value

Reply:
Never mind - after researching, I see you fixed that with `Option<>` (which I thought did something different).